src/classes/ADDR_Account_TDTM.cls

```diff
@@ -47,124 +70,183 @@ public class ADDR_Account_TDTM extends TDTM_Runnable {
+        ADDR_Addresses_UTIL.NonDupeAddrs nonDupeAddrs = ADDR_Addresses_UTIL.getNonDuplicateAddresses(ListAddr, false);
+        nonDupeAddrs.performDml();
+                
+        //Update child Contacts of those Accounts whose Addreses were just inserted, if they are in a Household
+        if(nonDupeAddrs.newAddrs.size() > 0) {
+	        updateChildContacts(nonDupeAddrs.newAddrs, dmlWrapper);
+        }
+        
+        //Update child Contacts of those Accounts whose Addreses were just updated, if they are in a Household
+        if(nonDupeAddrs.updatedAddrs.size() > 0) {
+            updateChildContacts(nonDupeAddrs.updatedAddrs, dmlWrapper);
+        }
+    }
+    
+    private void updateChildContacts(List<Address__c> newUpdatedAddrs, DmlWrapper dmlWrapper) {
+        //Get map of Accounts to their new/updated addresses
+        Set<ID> newUpdatedAddrsParentAccIDs = new Set<ID>();
+        Map<ID, Address__c> newUpdatedAddrsParentAccIdToAddr = new Map<ID, Address__c>();
+        for(Address__c addr : newUpdatedAddrs) {
+            if(addr.Parent_Account__c != null) {
+                //Putting new address with parent Account in map.
+                newUpdatedAddrsParentAccIDs.add(addr.Parent_Account__c);
+                newUpdatedAddrsParentAccIdToAddr.put(addr.Parent_Account__c, addr);  
+            }
+        }
+        
+        //We need to re-query because we are in the "after" part of the trigger. We cannot change it to before because
+        //we need the Account ID to have a value when populating the addr.Parent_Account__c field above.
+        List<Account> parentAccs = Database.query(ADDR_Addresses_UTIL.getParentAccsWithChildrenQuery() + ':newUpdatedAddrsParentAccIDs');
+                
+        for(Account acc : parentAccs) {
+            Address__c childAddr = newUpdatedAddrsParentAccIdToAddr.get(acc.Id);
+            //uncheck the old default address
+            if(childAddr.Default_Address__c)
+                ADDR_Addresses_UTIL.uncheckDefaultOtherAddrs(childAddr, acc.Addresses__r, dmlWrapper);
+            //link the account to the new default
+            if(childAddr != null) {
+                acc.Current_Address__c = childAddr.Id;
+                dmlWrapper.objectsToUpdate.add(acc);
+            }
+            //copy address info to child contacts, if it's a household and they don't have override
+            if(acc.RecordTypeId == UTIL_Describe.getHhAccRecTypeID()) {
+                for(Contact contact : acc.Contacts) {
+                    if(!contact.is_Address_Override__c) {
+                        ADDR_Addresses_UTIL.copyAddressStdSObj(acc, 'Billing', contact, 'Mailing');
+                        contact.Current_Address__c = childAddr.Id;
+                        dmlWrapper.objectsToUpdate.add(contact);
+                    }
+                }
+            }
+        }
+    }
+    
+    private void addrInfoDeleted(List<Account> accsAddrInfoCleared, DmlWrapper dmlWrapper) {
```

Comment:
apexdoc header on this would be useful to document its intent.  I had to read the code to understand what it wa trying to do.